kobo/apps/openrosa/apps/viewer/models/data_dictionary.py

```diff
@@ -157,9 +157,8 @@ def add_instances(self):
     def save(self, *args, **kwargs):
         if self.xls:
             survey = create_survey_from_xls(self.xls)
-            survey.update({
-                'name': survey.id_string,
-            })
+            if not survey.name or survey.name == 'None':
```

Comment:
Hi @rajpatel24, thanks for working on this! I'm curious: were there conditions where the string `None` gets passed in as the `name`? It seems a bit surprising to me

Reply:
Hi @jnm, When the form lacked a 'name' setting, the name was being set as the string 'None'.

Previously, we ignored the name setting value, whether it was 'None' or an actual string, and always assigned survey.id_string as the default name.

Now, if a valid name string is present, we leave it unchanged. If the name setting is absent, we assign survey.id_string as the default name.

Reply:
Sorry to hassle on this, but I wanted to know where the string `None` was coming from. It bothers me because someone could have their XLSForm `settings` sheet look like this:
| name |
| - |
| None|

It would be weird, but it'd be valid, and we wouldn't be able to tell the difference between a user-provided `None` and the one that gets generated when the `settings` sheet lacks a `name` altogether.

I did some digging and found that the `None` string is set by pyxform:
* It's a string because of this: https://github.com/XLSForm/pyxform/blob/9d20bfec5128f53ec0d555f7187f3bc4881f667e/pyxform/xls2json.py#L396-L397
* `None` itself comes from `default_name=None` in https://github.com/XLSForm/pyxform/blob/9d20bfec5128f53ec0d555f7187f3bc4881f667e/pyxform/builder.py#L327
* `default_name` gets overridden by the `name` value in the `settings` sheet of the XLSForm here: https://github.com/XLSForm/pyxform/blob/9d20bfec5128f53ec0d555f7187f3bc4881f667e/pyxform/xls2json.py#L469-L470

I see that even though `create_survey_from_xls()` has a handy `default_name` parameter, `create_survey_from_xls(self.xls, default_name=self.id_string)` won't work because our logic relies on `create_survey_from_xls()` pulling the `id_string` out of the XLS file in the first place. This is a historical artifact; nowadays, we're always the ones generating the XLS file internally¹, and [we always know the `id_string` in advance](https://github.com/kobotoolbox/kpi/blob/8564da5c63fdd1aa9a6e0b3335ec2fe08f81e492/kpi/deployment_backends/openrosa_backend.py#L129-L139).

Let's do a hack, though, instead of blowing up the scope of this PR. Please invent some constant that's unlikely ever to be a user-provided value, like `PLACEHOLDER_NAME = '__kobo_placeholder_name_value__'`, call `survey = create_survey_from_xls(self.xls, default_name=PLACEHOLDER_NAME)`, and then overwrite `survey.name` only if it's equal to `PLACEHOLDER_NAME`.

¹ Even when someone uploads their own XLSForm, we convert that to JSON, and then we convert it back to XLS again when deploying.